connector_test.go

```diff
@@ -47,3 +47,25 @@ func TestConnector(t *testing.T) {
 		t.Fatalf("Missing driver")
 	}
 }
+
+func TestConnectorWithMissingConfig(t *testing.T) {
+	conn := snowflakeConn{}
+	mock := noopTestDriver{conn: &conn}
+	config := Config{
+		User:     "u",
+		Password: "p",
+		Account:  "",
+	}
+	expectedErr := errEmptyAccount()
+
+	connector := NewConnector(&mock, config)
+	_, err := connector.Connect(context.Background())
+	assertNotNilF(t, err, "the connection should have failed due to empty account.")
+	driverErr, ok := err.(*SnowflakeError)
+	if !ok {
+		t.Fatalf("Snowflake error is expected. err: %v", err.Error())
+	}
+	if driverErr.Number != expectedErr.Number || driverErr.Message != expectedErr.Message {
```

Comment:
Can we split it into separate assertions?

Reply:
Updated.